python/src/pyarrow/adapters/pandas.cc

```diff
@@ -628,7 +628,6 @@ class ArrowDeserializer {
     PyAcquireGIL lock;
 
     // Zero-Copy. We can pass the data pointer directly to NumPy.
-    Py_INCREF(py_ref_);
```

Comment:
I'm surprised by this -- `PyArray_SetBaseObject` claims to steal a reference (https://docs.scipy.org/doc/numpy-dev/reference/c-api.array.html#c.PyArray_SetBaseObject), so if we are not the memory owner's reference count, then it could get garbage collected while the NumPy array is still alive. I believe you that it's leaking memory. To really be sure about this, we may wish to expose the `ConvertArrayToPandas` method in `pyarrow.Array.to_pandas` and then examine the reference counts before and after the conversion.